### PR TITLE
ADAPT-3957: Accommodation changes and add meal options to Itinerary item

### PR DESCRIPTION
**Pull request by yvonnetangsu**
# READY FOR REVIEW

# Summary
- Change to accommodation/hotel and add meal options field in Itinerary items

# Review By (Date)
- When does this need to be reviewed by?

# Criticality
- How critical is this PR on a 1-10 scale? Also see [Severity Assessment](https://stanfordits.atlassian.net/browse/D8CORE-1705).
- E.g., it affects one site, or every site and product?

# Review Tasks

## Setup tasks and/or behavior to test

1. https://deploy-preview-230--stanford-alumni.netlify.app/travel-study/trips/botswana-south-africa-and-zambia#itinerary-section
2. Look at the itinerary steps
3. Note changes to Accommodation and added "Included meals" field
4. Check that it looks like this
Screen Shot 2021-10-29 at 12 51 57 PM
5. scroll down to see other steps. If only 1 meal is selected it should say, eg, "Included meal: Lunch"; if 2 meals are selected, it should say "Included meals: Breakfast and lunch"; If all 3 are included, it should say "Included meals: Breakfast, lunch and dinner"
6. Only things called out below are actually new. The other changes shown in the diff are just lines getting shifted down.

## Front End Validation
- [ ] Is the markup using the appropriate semantic tags and passes HTML validation?
- [ ] Cross-browser testing has been performed?
- [ ] Automated accessibility scans performed?
- [ ] Manual accessibility tests performed?
- [ ] Design is approved by @ user?

### Code
- [ ] Are the naming conventions following our standards?
- [ ] Does the code have sufficient inline comments?
- [ ] Is there anything in this code that would be hidden or hard to discover through the UI?
- [ ] Are there any [code smells](https://blog.codinghorror.com/code-smells/)?
- [ ] Are tests provided? eg (unit, behat, or codeception)

### Code security
- [ ] Are all [forms properly sanitized](https://www.drupal.org/docs/8/security/drupal-8-sanitizing-output)?
- [ ] Any obvious [security flaws or new areas for attack](https://www.drupal.org/docs/8/security)?

## General
- [ ] Is there anything included in this PR that is not related to the problem it is trying to solve?
- [ ] Is the approach to the problem appropriate?

# Affected Projects or Products
- Does this PR impact any particular projects, products, or modules?

# Associated Issues and/or People
- JIRA ticket(s)
- Other PRs
- Any other contextual information that might be helpful (e.g., description of a bug that this PR fixes, new functionality that it adds, etc.)
- Anyone who should be notified? (`@mention` them here)

# Resources
- [AMP Tool](https://stanford.levelaccess.net/index.php)
- [Accessibility Manual Test Script](https://docs.google.com/document/d/1ZXJ9RIUNXsS674ow9j3qJ2g1OAkCjmqMXl0Gs8XHEPQ/edit?usp=sharing)
- [HTML Validator](https://validator.w3.org/)
- [Browserstack](https://live.browserstack.com/dashboard) and link to [Browserstack Credentials](https://asconfluence.stanford.edu/confluence/display/SWS/External+Account+Credentials)

### CODE REVIEW

File: src/components/composite/itineraryItem.js [hunk 1]
```diff
@@ -20,76 +13,119 @@ const ItineraryItem = ({
     startDate,
     endDate,
     icon,
-    location,
+    hotel,
+    meals = [],
```
Comment (yvonnetangsu) [inline src/components/composite/itineraryItem.js:17]: change prop name from `location` to `hotel`, and added `meals`

File: src/components/composite/itineraryItem.js [hunk 2]
```diff
@@ -20,76 +13,121 @@ const ItineraryItem = ({
...
-              aria-hidden="true"
+}) => {
+  // First take all but the last selected meal options and joins them together with a comma.
+  // Joins that string and the last element with "and" if there are at least two meal options selected.
+  const mealsString = [meals.slice(0, -1).join(', '), meals.slice(-1)[0]].join(
+    meals.length < 2 ? '' : ' and '
+  );
+
+  const mealsFormatted =
+    mealsString.charAt(0).toUpperCase() + mealsString.slice(1);
```
Comment (yvonnetangsu) [inline src/components/composite/itineraryItem.js:29]: Format the included meals display

File: src/components/composite/itineraryItem.js [hunk 3]
```diff
@@ -20,76 +13,121 @@ const ItineraryItem = ({
...
+                    />
+                  </div>
+                  <span className="su-leading-display">
+                    {`Included meal${meals.length > 1 ? 's' : ''}:
+                    ${mealsFormatted}`}
+                  </span>
+                </FlexBox>
+              )}
+            </div>
+          )}
```
Comment (yvonnetangsu) [inline src/components/composite/itineraryItem.js:109]: The hotel/meal part that has been modified in this PR

File: src/components/composite/itineraryItem.js [hunk 4]
```diff
@@ -20,76 +13,121 @@ const ItineraryItem = ({
...
-          <FlexBox direction="row" alignItems="start" className="su-rs-mt-1">
-            <LocationMarkerIcon
-              className="su-inline-block su-flex-shrink-0 su-mt-2 su-mr-03em su-w-[1em]"
-              aria-hidden="true"
+}) => {
+  // First take all but the last selected meal options and joins them together with a comma.
+  // Joins that string and the last element with "and" if there are at least two meal options selected.
+  const mealsString = [meals.slice(0, -1).join(', '), meals.slice(-1)[0]].join(
+    meals.length < 2 ? '' : ' and '
+  );
```
Comment (mattanglin) [inline src/components/composite/itineraryItem.js:26]: ✨

**Review by mattanglin — APPROVED** [review]
🔥 🍽️  🏨

**Pull request merged**